packages/ui-extensions/src/types/Button.ts

```diff
@@ -0,0 +1,58 @@
+export interface BaseButtonProps {
+  /**
+   * Destination to link to, renders a Link
+   */
+  to?: string;
+  /**
+   * The type of button that will be rendered.
+   *
+   * `primary`: button used for main actions or green-path. Ex: Continue to next step, Discount field
+   * `secondary`: button used for secondary actions not blocking user progress. Ex: Download Shop app
+   * `plain`: Renders a button that visually looks like a Link
+   * @default 'primary'*/
+  kind?: 'primary' | 'secondary' | 'plain';
+  /**
+   *  Specifies the color of the Button. The button will keep the style of the chodsen `kind`,
+   *  but replace its color according to the appearance.
+   *
+   * `monochrome`: button will inherit the color of its parent
+   * `critical`: button will take inherit the color of the Critical color group (in Checkout)
+   *  		 and map to 'destructive' (in Admin/Polaris). Typically used for destructive actions.
+   *
+   *  Excludes 'success' | 'warning' since buttons in a Banner will be `monochrome`
+   *  and use the default colors set by the Banner.
+   */
+  appearance?: 'monochrome' | 'critical';
+  /** Changes the size of the button
+   * @default 'base'
+   */
+  size?: 'base' | 'large';
+  /**
+   * Whether the button should fill all available inline space.
+   * */
+  fill?: boolean;
```

Comment:
We do have this in the button props, but I am not sure it fits with some of the more recent APIs we've done. In a lot of those APIs, we have left ourselves more room by defining `blockSize`/ `inlineSize` props with union types, which I expect we'd use for `Button` too when we do a polish pass on our components (e.g., `<Button inlineSize="fill" />`). I think this impacts `size` as well, which I think will be confusing if we have `blockSize`/ `inlineSize` as well. cc/ @jffortier @jun-shop

Reply:
I think I'm just pretty hesitant to commit to `boolean` props unless it is clear there will never be more than two options. If we choose `fill` for this, and later want a prop that is able to restrict the max size of the button or something similar to that, we are signed up for having a mix of attributes that do similar, but different, things. In contrast, using `inlineSize="fill"` here gives us tons of flexibility for the future to add more semantic sizes, or to allow for arbitrary sizes.

Don't feel super strongly about this one, but other than `disabled`, I haven't seen many props that are truly boolean in the long run.

Reply:
I'm good with `inlineSize="fill"`. We can make that change in Checkout.